daffodil-test/src/test/resources/org/apache/daffodil/section17/calc_value_properties/computedLengthFields.tdml

```diff
@@ -0,0 +1,214 @@
+
+        This creates a problem, as when unparsing, the length is computed using the messageLength
+        element, but the messageLength element is computed based on the length of the payload.
+
+        To avoid this circular definition, we compute the messageLength (unparsing) from
+        the payloadContents element. This is an artificial tier of element that must exist simply
+        to eliminate this circularity.
+        -->
+        <element name="payload" dfdl:lengthKind="explicit" dfdl:length="{ ../messageLength - 8 }">
+          <complexType>
+            <sequence>
+              <element name="payloadContents">
+                <complexType>
+                  <sequence>
+                    <!--
+                    The contents of the message payload go here, inside the payloadContents element.
+                    -->
+                    <element name="name" type="ex:pString"/>
+                    <element name="address" type="ex:pString"/>
+                  </sequence>
+                </complexType>
+              </element>
+            </sequence>
+          </complexType>
+        </element>
+      </sequence>
+    </complexType>
+
+    <!-- As a "control" for this experiment. Let's create a fixed-length string -->
+    <simpleType name="fString" dfdl:lengthKind="explicit">
+      <restriction base="xs:string"/>
+    </simpleType>
+
+    <element name="message2" type="ex:messageWithFixedLengthStringsInPayloadType"/>
+
+    <complexType name="messageWithFixedLengthStringsInPayloadType">
+      <sequence>
+        <element name="messageLength" type="unsignedInt"
+                 dfdl:outputValueCalc="{
+                     dfdl:valueLength( ../payload/payloadContents, 'bytes') + 8
+                   }"/>
+        <element name="num" type="xs:unsignedInt"/>
+        <element name="payload" dfdl:lengthKind="explicit" dfdl:length="{ ../messageLength - 8 }">
+          <complexType>
+            <sequence>
+              <element name="payloadContents">
+                <complexType>
+                  <sequence>
+                    <element name="name" type="ex:fString" dfdl:length="5"/>
+                    <element name="address" type="ex:fString" dfdl:length="6"/>
+                  </sequence>
+                </complexType>
+              </element>
+            </sequence>
+          </complexType>
+        </element>
+      </sequence>
+    </complexType>
+
+
+  </tdml:defineSchema>
+
+  <parserTestCase name="computedLengthAroundPrefixedLengths1p" model="s"
+                  roundTrip="none"
+                  description="Shows that we can parse data with an overall message length,
+    and payload containing two prefixed-length strings.">
+    <document>
+      <documentPart type="byte"><![CDATA[
+      0000 0017
+      0000 1ABF
+      0005 46 6F 6F 62 79
+      0006 4E 6F 6F 62 79 4E
+    ]]></documentPart>
+    </document>
+    <infoset>
+      <dfdlInfoset>
+        <ex:message xmlns="">
```

Comment:
Notice another redefine of xmlns here also. Needed so that the inner children elements are interpreted as no-namespace (corresponding to elementFormDefault 'unqualified' in the schema definition).